src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("There are no such index: " + index);
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        addValueByIndex(newNode, index, value);
```

Comment:
why did you moved this to separate method? you dont use it nowhere else

Reply:
i mean 40 line

Reply:
I moved this part of code to a separate method to make code more readable.

Reply:
i`m gonna aprove, but i`m still not shure, so let`s wait for mentors reviev